### ISSUE TITLE
Cannot mysqldef twice with CHECK constraint written

### ISSUE THREAD

**Issue by kotahashihama** [k0kubun/sqldef#204]
CHECK constraint can be correctly processed when trying the first `mysqldef` but the second or later it'll be failed with same SQL.

## Input SQL

```sql
CREATE TABLE `books` (
  `id` int(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
  `type` VARCHAR(255) NOT NULL COMMENT 'manga, novel',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `deleted_at` datetime DEFAULT null,
  CONSTRAINT `books_type_chk` CHECK (`type` IN ('manga', 'novel'))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC;
```

## Current output

```
-- Apply --
CREATE TABLE `books` (
  `id` int(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
  `type` VARCHAR(255) NOT NULL COMMENT 'manga, novel',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `deleted_at` datetime DEFAULT null,
  CONSTRAINT `books_type_chk` CHECK (`type` IN ('manga', 'novel'))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC;
-- Apply --
CREATE TABLE `books` (
  `id` int(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
  `type` VARCHAR(255) NOT NULL COMMENT 'manga, novel',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `deleted_at` datetime DEFAULT null,
  CONSTRAINT `books_type_chk` CHECK (`type` IN ('manga', 'novel'))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC;
```

## Expected output

```
found syntax error when parsing DDL "CREATE TABLE `books` (
  `id` int unsigned NOT NULL AUTO_INCREMENT,
  `type` varchar(255) COLLATE utf8mb4_bin NOT NULL COMMENT 'manga, novel',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `deleted_at` datetime DEFAULT NULL,
  PRIMARY KEY (`id`),
  CONSTRAINT `books_type_chk` CHECK ((`type` in (_utf8mb4'manga',_utf8mb4'novel')))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC": syntax error at position 416 near 'manga'
found syntax error when parsing DDL "CREATE TABLE `books` (
  `id` int unsigned NOT NULL AUTO_INCREMENT,
  `type` varchar(255) COLLATE utf8mb4_bin NOT NULL COMMENT 'manga, novel',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `deleted_at` datetime DEFAULT NULL,
  PRIMARY KEY (`id`),
  CONSTRAINT `books_type_chk` CHECK ((`type` in (_utf8mb4'manga',_utf8mb4'novel')))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC": syntax error at position 416 near 'manga'
make: *** [local_db_migration] Error 1
```

**Comment by kotahashihama** [comment]
The below with column definition and CHECK constraint one-lined and in REGEXP makes no error, though CHECK constraint removed after the second or later.


## Input SQL

```sql
CREATE TABLE `books` (
  `id` int(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
  `type` VARCHAR(255) NOT NULL CONSTRAINT `books_type_chk` CHECK (`type` REGEXP '^(manga|novel)$') COMMENT 'manga, novel',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `deleted_at` datetime DEFAULT null
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC;
```


## Output by the first `mysqldef`

```
-- Apply --
CREATE TABLE `books` (
  `id` int(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
  `type` VARCHAR(255) NOT NULL CONSTRAINT `books_type_chk` CHECK (`type` REGEXP '^(manga|novel)$') COMMENT 'manga, novel',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `deleted_at` datetime DEFAULT null
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC;
-- Apply --
CREATE TABLE `books` (
  `id` int(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
  `type` VARCHAR(255) NOT NULL CONSTRAINT `books_type_chk` CHECK (`type` REGEXP '^(manga|novel)$') COMMENT 'manga, novel',
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `deleted_at` datetime DEFAULT null
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC;
```

CHECK constraint enabled.


## Output by the second `mysqldef`

```
-- Apply --
ALTER TABLE `books` CHANGE COLUMN `type` `type` varchar(255) NOT NULL COMMENT 'manga, novel' CHECK (`type` regexp '^(manga|novel)$');
ALTER TABLE `books` DROP CONSTRAINT `books_type_chk`;
-- Apply --
ALTER TABLE `books` CHANGE COLUMN `type` `type` varchar(255) NOT NULL COMMENT 'manga, novel' CHECK (`type` regexp '^(manga|novel)$');
ALTER TABLE `books` DROP CONSTRAINT `books_type_chk`;
```

CHECK constraint disabled.


## Output by the third or more `mysqldef`

```
-- Apply --
ALTER TABLE `books` CHANGE COLUMN `type` `type` varchar(255) NOT NULL COMMENT 'manga, novel' CHECK (`type` regexp '^(manga|novel)$');
-- Apply --
ALTER TABLE `books` CHANGE COLUMN `type` `type` varchar(255) NOT NULL COMMENT 'manga, novel' CHECK (`type` regexp '^(manga|novel)$');
```

CHECK constraint disabled.

**Comment by k0kubun** [comment]
I started to take a look at this. I noticed that MySQL ignores CHECK constraints when you use `show create table books;`. It's currently blocking me from fixing it. If you know how to list CHECK constraints in MySQL by using information_schema or anything, please let me know.

**Comment by k0kubun** [comment]
Oh, sorry, I was testing it with a table created by 5.7, but if I create a table with 8.0 and check it, it seems to work. So I can probably fix it.

By the way, what MySQL server version are you using? I'll consider adding the version to the CI matrix.

**Comment by kotahashihama** [comment]
> what MySQL server version are you using?

The MySQL version is 8.0.27.

**Comment by kotahashihama** [comment]
@k0kubun 
> what MySQL server version are you using?

The MySQL version is 8.0.27.

**Comment by k0kubun** [comment]
I fixed the reported test cases in v0.11.37.
